src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/RetrieveMinimalOrganisationsInfoIntegrationTest.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+    @Test
+    //AC:5
+    public void should_fail_to_retrieve_organisations_info_with_404_with_correct_roles_and_status_pending() throws JsonProcessingException {
+        inviteUser(false);
+        Map<String, Object> errorResponseMap = (Map<String, Object>) professionalReferenceDataClient.retrieveOrganisationsWithMinimalInfo(
+                userIdentifier, puiCaa, PENDING.toString(), ErrorResponse.class);
+        validateErrorResponse(errorResponseMap, HttpStatus.NOT_FOUND, EMPTY_RESULT_DATA_ACCESS.getErrorMessage(), STATUS_PARAM_INVALID_MESSAGE);
+    }
+
+    @Test
+    //AC:6
+    public void should_fail_to_retrieve_organisations_info_with_404_with_correct_roles_and_status_not_passed() throws JsonProcessingException {
+        inviteUser(false);
+        Map<String, Object> errorResponseMap = (Map<String,Object>) professionalReferenceDataClient.retrieveOrganisationsWithMinimalInfo(
+                userIdentifier, puiCaa, null, ErrorResponse.class);
+        validateErrorResponse(errorResponseMap, HttpStatus.NOT_FOUND, EMPTY_RESULT_DATA_ACCESS.getErrorMessage(), STATUS_PARAM_INVALID_MESSAGE);
+    }
+
+    public void validateErrorResponse(Map<String, Object> errorResponseMap, HttpStatus expectedStatus, String expectedErrorMessage, String expectedErrorDescription) {
+        assertThat(errorResponseMap.get("http_status")).isEqualTo(expectedStatus);
+        ErrorResponse errorResponse = (ErrorResponse)errorResponseMap.get("response_body");
+        assertThat(errorResponse.getErrorDescription()).isEqualTo(expectedErrorDescription);
+        assertThat(errorResponse.getErrorMessage()).isEqualTo(expectedErrorMessage);
+    }
+
+    public void setUpTestData() {
+        createActiveOrganisation1();
+        createActiveOrganisation2();
+        createPendingOrganisation1();
+        createPendingOrganisation2();
+        inviteUser(true);
+    }
+
+    public void inviteUser(boolean useExistingOrg) {
+
+        if (!useExistingOrg) {
+            createActiveOrganisation1();
+        }
+        userProfileCreateUserWireMock(HttpStatus.CREATED);
+        NewUserCreationRequest newUserCreationRequest = inviteUserCreationRequest(randomAlphabetic(5) + "@somedomain.com", getValidRoleList());
+        Map<String, Object> newUserResponse = professionalReferenceDataClient.addUserToOrganisation(orgIdentifier1, newUserCreationRequest, hmctsAdmin);
+        userIdentifier = (String)newUserResponse.get("userIdentifier");
+    }
+
+    public String createActiveOrganisation1() {
+        userProfileCreateUserWireMock(HttpStatus.CREATED);
+        String orgName1 = randomAlphabetic(7);
+        orgIdentifier1 = createAndActivateOrganisationWithGivenRequest(someMinimalOrganisationRequest().name(orgName1).sraId(randomAlphabetic(10)).build());
+        organisationEntityResponse1 = new OrganisationMinimalInfoResponse(orgName1, orgIdentifier1);
+        activeOrgs.put(orgIdentifier1, organisationEntityResponse1);
```

Comment:
no need to create two methods here ..its already using randomAlphabetic(7) so every time it creates new name. so please delete other and make use two time if you want create two organisations.